// @workday/canvas-kit-react-button/dist/commonjs/lib/HighlightButton.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var canvas_kit_react_common_1 = require("@workday/canvas-kit-react-common");
var parts_1 = require("./parts");
var getHighlightButtonColors = function (_a) {
    var themePrimary = _a.canvas.palette.primary;
    return ({
        default: {
            background: canvas_kit_react_core_1.colors.soap200,
            border: canvas_kit_react_core_1.colors.soap200,
            icon: themePrimary.dark,
            label: themePrimary.dark,
        },
        hover: {
            background: canvas_kit_react_core_1.colors.soap400,
            border: 'transparent',
            icon: themePrimary.dark,
            label: themePrimary.dark,
        },
        active: {
            background: canvas_kit_react_core_1.colors.soap500,
            border: 'transparent',
            icon: themePrimary.dark,
            label: themePrimary.dark,
        },
        focus: {
            background: canvas_kit_react_core_1.colors.soap200,
            border: 'transparent',
            icon: themePrimary.dark,
            label: themePrimary.dark,
        },
        disabled: {
            background: canvas_kit_react_core_1.colors.soap100,
            border: 'transparent',
            icon: canvas_kit_react_core_1.colors.soap600,
            label: canvas_kit_react_core_1.colors.licorice100,
        },
    });
};
var HighlightButton = function (_a) {
    var _b = _a.theme, theme = _b === void 0 ? canvas_kit_react_common_1.useTheme() : _b, _c = _a.size, size = _c === void 0 ? 'medium' : _c, buttonRef = _a.buttonRef, icon = _a.icon, children = _a.children, elemProps = __rest(_a, ["theme", "size", "buttonRef", "icon", "children"]);
    return (React.createElement(parts_1.ButtonContainer, __assign({ colors: getHighlightButtonColors(theme), size: size, ref: buttonRef }, elemProps),
        icon && React.createElement(parts_1.ButtonLabelIcon, { size: size, icon: icon }),
        React.createElement(parts_1.ButtonLabel, null, children)));
};
HighlightButton.Size = {
    Medium: 'medium',
    Large: 'large',
};
HighlightButton.Variant = undefined;
exports.default = HighlightButton;
