// @workday/canvas-kit-react-button/dist/commonjs/lib/ToolbarDropdownButton.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@emotion/core");
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var canvas_kit_react_common_1 = require("@workday/canvas-kit-react-common");
var canvas_kit_react_icon_1 = require("@workday/canvas-kit-react-icon");
var parts_1 = require("./parts");
var canvas_system_icons_web_1 = require("@workday/canvas-system-icons-web");
var containerStyles = {
    padding: canvas_kit_react_core_1.spacing.zero,
    minWidth: canvas_kit_react_core_1.spacing.l,
    width: 'auto',
    height: canvas_kit_react_core_1.spacing.l,
    borderRadius: canvas_kit_react_core_1.borderRadius.m,
    '& .wd-icon': {
        display: 'inline-block',
        verticalAlign: 'middle',
        width: 20,
        height: 20,
    },
    '& .wdc-toolbar-dropdown-btn-arrow': {
        margin: '0 2px 0 0',
    },
    '& .wdc-toolbar-dropdown-btn-custom-icon': {
        marginLeft: "" + canvas_kit_react_core_1.spacing.xxxs,
        marginRight: 0,
        width: 18,
    },
};
var ToolbarDropdownButton = function (_a) {
    var _b = _a.theme, theme = _b === void 0 ? canvas_kit_react_common_1.useTheme() : _b, buttonRef = _a.buttonRef, iconArialabel = _a["aria-label"], icon = _a.icon, children = _a.children, elemProps = __rest(_a, ["theme", "buttonRef", 'aria-label', "icon", "children"]);
    return (core_1.jsx(parts_1.ButtonContainer, __assign({ colors: getToolbarDropdownButtonColors(theme), ref: buttonRef, extraStyles: containerStyles, "aria-label": iconArialabel }, elemProps),
        icon ? (core_1.jsx(canvas_kit_react_icon_1.SystemIcon, { className: 'wdc-toolbar-dropdown-btn-custom-icon', icon: icon })) : (children),
        core_1.jsx(canvas_kit_react_icon_1.SystemIcon, { className: 'wdc-toolbar-dropdown-btn-arrow', icon: canvas_system_icons_web_1.chevronDownSmallIcon })));
};
exports.default = ToolbarDropdownButton;
var getToolbarDropdownButtonColors = function (theme) {
    return {
        default: {
            icon: canvas_kit_react_core_1.colors.licorice200,
        },
        hover: {
            icon: canvas_kit_react_core_1.colors.licorice500,
            background: canvas_kit_react_core_1.colors.soap300,
        },
        active: {
            icon: canvas_kit_react_core_1.colors.licorice500,
            background: canvas_kit_react_core_1.colors.soap500,
        },
        focus: {
            icon: canvas_kit_react_core_1.colors.licorice200,
            focusRing: canvas_kit_react_common_1.focusRing({ width: 2, separation: 0 }, theme),
            background: 'transparent',
        },
        disabled: {
            icon: canvas_kit_react_core_1.colors.soap600,
            background: 'transparent',
        },
    };
};
