// @workday/canvas-kit-react-button/dist/es6/lib/HighlightButton.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { colors } from '@workday/canvas-kit-react-core';
import { useTheme, } from '@workday/canvas-kit-react-common';
import { ButtonContainer, ButtonLabel, ButtonLabelIcon } from './parts';
var getHighlightButtonColors = function (_a) {
    var themePrimary = _a.canvas.palette.primary;
    return ({
        default: {
            background: colors.soap200,
            border: colors.soap200,
            icon: themePrimary.dark,
            label: themePrimary.dark,
        },
        hover: {
            background: colors.soap400,
            border: 'transparent',
            icon: themePrimary.dark,
            label: themePrimary.dark,
        },
        active: {
            background: colors.soap500,
            border: 'transparent',
            icon: themePrimary.dark,
            label: themePrimary.dark,
        },
        focus: {
            background: colors.soap200,
            border: 'transparent',
            icon: themePrimary.dark,
            label: themePrimary.dark,
        },
        disabled: {
            background: colors.soap100,
            border: 'transparent',
            icon: colors.soap600,
            label: colors.licorice100,
        },
    });
};
var HighlightButton = function (_a) {
    var _b = _a.theme, theme = _b === void 0 ? useTheme() : _b, _c = _a.size, size = _c === void 0 ? 'medium' : _c, buttonRef = _a.buttonRef, icon = _a.icon, children = _a.children, elemProps = __rest(_a, ["theme", "size", "buttonRef", "icon", "children"]);
    return (React.createElement(ButtonContainer, __assign({ colors: getHighlightButtonColors(theme), size: size, ref: buttonRef }, elemProps),
        icon && React.createElement(ButtonLabelIcon, { size: size, icon: icon }),
        React.createElement(ButtonLabel, null, children)));
};
HighlightButton.Size = {
    Medium: 'medium',
    Large: 'large',
};
HighlightButton.Variant = undefined;
export default HighlightButton;
