// @workday/canvas-kit-react-button/dist/es6/lib/OutlineButton.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { colors } from '@workday/canvas-kit-react-core';
import { focusRing, useTheme, } from '@workday/canvas-kit-react-common';
import { OutlineButtonVariant, ButtonSize } from './types';
import { ButtonContainer, ButtonLabel, ButtonLabelData, ButtonLabelIcon } from './parts';
var OutlineButton = function (_a) {
    var _b = _a.theme, theme = _b === void 0 ? useTheme() : _b, _c = _a.variant, variant = _c === void 0 ? OutlineButtonVariant.Secondary : _c, _d = _a.size, size = _d === void 0 ? 'medium' : _d, buttonRef = _a.buttonRef, dataLabel = _a.dataLabel, icon = _a.icon, children = _a.children, elemProps = __rest(_a, ["theme", "variant", "size", "buttonRef", "dataLabel", "icon", "children"]);
    return (React.createElement(ButtonContainer, __assign({ colors: getOutlineButtonColors(variant, theme), size: size, ref: buttonRef }, elemProps),
        icon && size !== 'small' && React.createElement(ButtonLabelIcon, { size: size, icon: icon }),
        React.createElement(ButtonLabel, null, children),
        dataLabel && size !== 'small' && React.createElement(ButtonLabelData, null, dataLabel)));
};
OutlineButton.Variant = OutlineButtonVariant;
OutlineButton.Size = ButtonSize;
export default OutlineButton;
export var getOutlineButtonColors = function (variant, theme) {
    var themePrimary = theme.canvas.palette.primary;
    switch (variant) {
        case OutlineButtonVariant.Primary:
            return {
                default: {
                    background: 'transparent',
                    border: themePrimary.main,
                    icon: themePrimary.main,
                    label: themePrimary.main,
                },
                hover: {
                    background: themePrimary.main,
                    icon: themePrimary.contrast,
                    label: themePrimary.contrast,
                },
                active: {
                    background: themePrimary.dark,
                    border: themePrimary.dark,
                    icon: themePrimary.contrast,
                    label: themePrimary.contrast,
                },
                focus: {
                    background: themePrimary.main,
                    icon: themePrimary.contrast,
                    label: themePrimary.contrast,
                },
                disabled: {
                    background: 'transparent',
                    border: colors.soap500,
                    icon: colors.soap600,
                    label: colors.licorice100,
                },
            };
        case OutlineButtonVariant.Secondary:
        default:
            return {
                default: {
                    background: 'transparent',
                    border: colors.soap500,
                    icon: colors.licorice200,
                    label: colors.blackPepper400,
                },
                hover: {
                    background: colors.licorice500,
                    border: colors.licorice500,
                    icon: themePrimary.contrast,
                    label: themePrimary.contrast,
                },
                active: {
                    background: colors.licorice600,
                    border: colors.licorice600,
                    icon: themePrimary.contrast,
                    label: themePrimary.contrast,
                },
                focus: {
                    background: colors.licorice500,
                    border: colors.licorice500,
                    icon: themePrimary.contrast,
                    label: themePrimary.contrast,
                },
                disabled: {
                    background: 'transparent',
                    border: colors.soap500,
                    icon: colors.soap600,
                    label: colors.licorice100,
                },
            };
        case OutlineButtonVariant.Inverse:
            return {
                default: {
                    background: 'transparent',
                    border: colors.frenchVanilla100,
                    icon: colors.frenchVanilla100,
                    label: colors.frenchVanilla100,
                },
                hover: {
                    background: colors.frenchVanilla100,
                    icon: colors.licorice500,
                    label: colors.blackPepper400,
                    labelData: colors.licorice300,
                },
                active: {
                    background: colors.soap300,
                    border: colors.soap300,
                    icon: colors.licorice500,
                    label: colors.blackPepper400,
                    labelData: colors.licorice300,
                },
                focus: {
                    background: colors.frenchVanilla100,
                    icon: colors.licorice500,
                    label: colors.blackPepper400,
                    labelData: colors.licorice300,
                    focusRing: focusRing({
                        separation: 2,
                        innerColor: 'currentColor',
                        outerColor: colors.frenchVanilla100,
                    }, theme),
                },
                disabled: {
                    background: 'transparent',
                    border: 'rgba(255, 255, 255, 0.75)',
                    icon: 'rgba(255, 255, 255, 0.75)',
                    label: 'rgba(255, 255, 255, 0.75)',
                    labelData: 'rgba(255, 255, 255, 0.75)',
                },
            };
    }
};
