// @workday/canvas-kit-react-button/dist/es6/lib/TextButton.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { type } from '@workday/canvas-kit-labs-react-core';
import { focusRing, useTheme } from '@workday/canvas-kit-react-common';
import { colors, spacing, borderRadius } from '@workday/canvas-kit-react-core';
import { TextButtonVariant, ButtonIconPosition, } from './types';
import { ButtonContainer, ButtonLabelIcon, ButtonLabel } from './parts';
var getTextButtonColors = function (variant, theme) {
    var themePrimary = theme.canvas.palette.primary;
    switch (variant) {
        case TextButtonVariant.Default:
        default:
            return {
                default: {
                    icon: themePrimary.main,
                    label: themePrimary.main,
                },
                hover: {
                    background: colors.soap200,
                    icon: themePrimary.dark,
                    label: themePrimary.dark,
                },
                active: {
                    background: colors.soap300,
                    icon: themePrimary.dark,
                    label: themePrimary.dark,
                },
                focus: {
                    icon: themePrimary.dark,
                    label: themePrimary.dark,
                    focusRing: focusRing({}, theme),
                },
                disabled: {
                    background: 'transparent',
                    icon: themePrimary.light,
                    label: themePrimary.light,
                },
            };
        case TextButtonVariant.Inverse:
            return {
                default: {
                    background: 'transparent',
                    icon: colors.frenchVanilla100,
                    label: colors.frenchVanilla100,
                },
                hover: {
                    background: colors.frenchVanilla100,
                    icon: colors.blackPepper400,
                    label: colors.blackPepper400,
                },
                active: {
                    background: colors.soap200,
                    icon: colors.blackPepper400,
                    label: colors.blackPepper400,
                },
                focus: {
                    background: colors.frenchVanilla100,
                    icon: colors.blackPepper400,
                    label: colors.blackPepper400,
                    focusRing: focusRing({
                        separation: 2,
                        inset: 'inner',
                        innerColor: 'currentColor',
                        outerColor: colors.frenchVanilla100,
                    }, theme),
                },
                disabled: {
                    background: 'transparent',
                    icon: 'rgba(255, 255, 255, 0.5)',
                    label: 'rgba(255, 255, 255, 0.5)',
                },
            };
    }
};
var containerStyles = {
    borderRadius: borderRadius.m,
    border: '0',
    padding: "0 " + spacing.xxs,
    minWidth: 'auto',
    '.wdc-text-button-label': {
        borderBottom: '2px solid transparent',
        paddingTop: '2px',
        transition: 'border-color 0.3s',
    },
    '&:hover:not([disabled]) .wdc-text-button-label': {
        borderBottomColor: 'currentColor',
    },
};
var TextButton = function (_a) {
    var _b = _a.theme, theme = _b === void 0 ? useTheme() : _b, _c = _a.variant, variant = _c === void 0 ? TextButtonVariant.Default : _c, _d = _a.size, size = _d === void 0 ? 'medium' : _d, _e = _a.iconPosition, iconPosition = _e === void 0 ? ButtonIconPosition.Left : _e, buttonRef = _a.buttonRef, children = _a.children, icon = _a.icon, allCaps = _a.allCaps, elemProps = __rest(_a, ["theme", "variant", "size", "iconPosition", "buttonRef", "children", "icon", "allCaps"]);
    var allContainerStyles = allCaps
        ? __assign(__assign(__assign(__assign({}, containerStyles), type.variant.caps), type.variant.button), { fontSize: size === 'medium' ? type.body.fontSize : undefined, letterSpacing: '.5px' }) : __assign(__assign({}, containerStyles), { fontSize: size === 'medium' ? type.body.fontSize : undefined });
    return (React.createElement(ButtonContainer, __assign({ colors: getTextButtonColors(variant, theme), ref: buttonRef, size: size, extraStyles: allContainerStyles }, elemProps),
        icon && iconPosition === ButtonIconPosition.Left && (React.createElement(ButtonLabelIcon, { size: size, iconPosition: iconPosition, icon: icon })),
        React.createElement(ButtonLabel, { className: "wdc-text-button-label" }, children),
        icon && iconPosition === ButtonIconPosition.Right && (React.createElement(ButtonLabelIcon, { size: size, iconPosition: iconPosition, icon: icon }))));
};
TextButton.IconPosition = ButtonIconPosition;
TextButton.Variant = TextButtonVariant;
TextButton.Size = {
    Small: 'small',
    Medium: 'medium',
};
export default TextButton;
