// @workday/canvas-kit-labs-react-select/dist/es6/lib/SelectBase.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React, { useLayoutEffect } from 'react';
import uuid from 'uuid/v4';
import { errorRing, styled, } from '@workday/canvas-kit-react-common';
import { colors, borderRadius, inputColors, spacingNumbers, type, spacing, } from '@workday/canvas-kit-react-core';
import { caretDownSmallIcon } from '@workday/canvas-system-icons-web';
import { SystemIcon } from '@workday/canvas-kit-react-icon';
import SelectMenu from './SelectMenu';
import SelectOption from './SelectOption';
import { scrollIntoViewIfNeeded } from './scrolling';
import { getCorrectedIndexByValue } from './utils';
export var buttonBorderWidth = 1;
export var buttonDefaultWidth = 280;
var menuIconSize = 24;
var buttonPadding = spacingNumbers.xxs - buttonBorderWidth;
var SelectButton = styled('button')(__assign(__assign({}, type.body), { border: buttonBorderWidth + "px solid " + inputColors.border, cursor: 'default', display: 'block', backgroundColor: inputColors.background, borderRadius: borderRadius.m, boxSizing: 'border-box', height: spacing.xl, outline: 'none', overflow: 'hidden', padding: buttonPadding, paddingRight: spacingNumbers.xxs + menuIconSize + buttonPadding, textAlign: 'left', textOverflow: 'ellipsis', transition: '0.2s box-shadow, 0.2s border-color', whiteSpace: 'nowrap', width: buttonDefaultWidth, '&::placeholder': {
        color: inputColors.placeholder,
    }, '&:disabled': {
        backgroundColor: inputColors.disabled.background,
        borderColor: inputColors.disabled.border,
        color: inputColors.disabled.text,
        '&::placeholder': {
            color: inputColors.disabled.text,
        },
    } }), function (_a) {
    var error = _a.error, menuVisibility = _a.menuVisibility, theme = _a.theme;
    var themedFocusOutlineColor = theme.canvas.palette.common.focusOutline;
    var buttonFocusStyles = {
        borderColor: themedFocusOutlineColor,
        boxShadow: "inset 0 0 0 1px " + themedFocusOutlineColor,
    };
    if (error === undefined) {
        return menuVisibility === 'closed' || menuVisibility === 'closing'
            ? {
                '&:focus:not([disabled])': __assign({}, buttonFocusStyles),
                '&:hover:not([disabled]):not(:focus)': {
                    borderColor: inputColors.hoverBorder,
                },
            }
            : __assign({}, buttonFocusStyles);
    }
    return __assign({}, errorRing(error, theme));
}, function (_a) {
    var grow = _a.grow;
    return grow && {
        width: '100%',
    };
});
var SelectMenuIcon = styled(SystemIcon)({
    position: 'absolute',
    top: spacing.xxxs,
    right: spacing.xxxs,
    padding: spacing.xxxs,
    pointerEvents: 'none',
    '& path': {
        transition: '100ms fill',
    },
});
var SelectInput = styled('input')({
    display: 'none',
});
var SelectWrapper = styled('div')({
    position: 'relative',
}, function (_a) {
    var grow = _a.grow;
    return ({
        display: grow ? 'block' : 'inline-block',
    });
}, function (_a) {
    var disabled = _a.disabled;
    return ({
        '&:hover .menu-icon path': {
            fill: disabled ? undefined : colors.licorice500,
        },
    });
});
var defaultRenderOption = function (option) {
    return React.createElement("div", null, option.label);
};
var SelectBase = function (_a) {
    var ariaLabelledBy = _a["aria-labelledby"], ariaRequired = _a["aria-required"], buttonRef = _a.buttonRef, disabled = _a.disabled, error = _a.error, _b = _a.focusedOptionIndex, focusedOptionIndex = _b === void 0 ? 0 : _b, grow = _a.grow, inputRef = _a.inputRef, _c = _a.menuPlacement, menuPlacement = _c === void 0 ? 'bottom' : _c, menuRef = _a.menuRef, _d = _a.menuVisibility, menuVisibility = _d === void 0 ? 'closed' : _d, onChange = _a.onChange, onKeyDown = _a.onKeyDown, onMenuBlur = _a.onMenuBlur, onMenuCloseOnEscape = _a.onMenuCloseOnEscape, onOptionSelection = _a.onOptionSelection, options = _a.options, _e = _a.renderOption, renderOption = _e === void 0 ? defaultRenderOption : _e, required = _a.required, _f = _a.shouldMenuAutoFlip, shouldMenuAutoFlip = _f === void 0 ? true : _f, _g = _a.shouldMenuAutoFocus, shouldMenuAutoFocus = _g === void 0 ? true : _g, value = _a.value, elemProps = __rest(_a, ['aria-labelledby', 'aria-required', "buttonRef", "disabled", "error", "focusedOptionIndex", "grow", "inputRef", "menuPlacement", "menuRef", "menuVisibility", "onChange", "onKeyDown", "onMenuBlur", "onMenuCloseOnEscape", "onOptionSelection", "options", "renderOption", "required", "shouldMenuAutoFlip", "shouldMenuAutoFocus", "value"]);
    var focusedOptionRef = React.useRef(null);
    var menuId = React.useState(function () { return "a" + uuid(); })[0];
    var renderOptions = function (renderOption) {
        var selectedOptionIndex = getCorrectedIndexByValue(options, value);
        return options.map(function (option, index) {
            var optionProps = __assign({ 'aria-disabled': option.disabled ? true : undefined, 'aria-selected': selectedOptionIndex === index ? true : undefined, error: error, focused: focusedOptionIndex === index, id: option.id, interactive: menuVisibility === 'opening' || menuVisibility === 'opened', key: option.id, optionRef: focusedOptionIndex === index ? focusedOptionRef : undefined, value: option.value }, (onOptionSelection
                ? {
                    onClick: function (event) {
                        event.preventDefault();
                        onOptionSelection(index);
                    },
                }
                : {}));
            var normalizedOption = __assign(__assign({}, option), { focused: optionProps.focused, selected: !!optionProps['aria-selected'] });
            return React.createElement(SelectOption, __assign({}, optionProps), renderOption(normalizedOption));
        });
    };
    useLayoutEffect(function () {
        var focusedOption = focusedOptionRef.current;
        if (focusedOption) {
            var animateId_1 = requestAnimationFrame(function () {
                scrollIntoViewIfNeeded(focusedOption, false);
            });
            return function () {
                cancelAnimationFrame(animateId_1);
            };
        }
        return undefined;
    }, [focusedOptionIndex]);
    useLayoutEffect(function () {
        var focusedOption = focusedOptionRef.current;
        if (focusedOption && (menuVisibility === 'opening' || menuVisibility === 'opened')) {
            var animateId_2 = requestAnimationFrame(function () {
                scrollIntoViewIfNeeded(focusedOption, true);
            });
            return function () {
                cancelAnimationFrame(animateId_2);
            };
        }
        return undefined;
    }, [menuVisibility]);
    var hasOptions = options.length > 0;
    var selectedOption = hasOptions ? options[getCorrectedIndexByValue(options, value)] : null;
    var selectedOptionLabel = selectedOption ? selectedOption.label : '';
    var selectedOptionValue = selectedOption ? selectedOption.value : '';
    return (React.createElement(SelectWrapper, { grow: grow, disabled: disabled },
        React.createElement(SelectButton, __assign({ "aria-expanded": menuVisibility !== 'closed' ? 'true' : undefined, "aria-haspopup": "listbox", "aria-controls": menuVisibility !== 'closed' ? menuId : undefined, disabled: disabled, error: error, grow: grow, menuVisibility: menuVisibility, onKeyDown: onKeyDown, onKeyUp: function (e) {
                e.preventDefault();
            }, ref: buttonRef, type: "button", value: selectedOptionValue }, elemProps), selectedOptionLabel),
        React.createElement(SelectInput, { onChange: onChange, ref: inputRef, type: "text", value: selectedOptionValue }),
        hasOptions && menuVisibility !== 'closed' && (React.createElement(SelectMenu, { "aria-activedescendant": options[focusedOptionIndex].id, "aria-labelledby": ariaLabelledBy, "aria-required": ariaRequired || required ? true : undefined, buttonRef: buttonRef, id: menuId, error: error, menuRef: menuRef, onBlur: onMenuBlur, onKeyDown: onKeyDown, onCloseOnEscape: onMenuCloseOnEscape, placement: menuPlacement, shouldAutoFlip: shouldMenuAutoFlip, shouldAutoFocus: shouldMenuAutoFocus, visibility: menuVisibility }, renderOptions(renderOption))),
        React.createElement(SelectMenuIcon, { className: "menu-icon", icon: caretDownSmallIcon, color: disabled ? colors.licorice100 : colors.licorice200, colorHover: disabled ? colors.licorice100 : colors.licorice500, size: menuIconSize })));
};
export default SelectBase;
