// @workday/canvas-kit-react-card/dist/commonjs/lib/Card.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var styled_1 = __importDefault(require("@emotion/styled"));
var is_prop_valid_1 = __importDefault(require("@emotion/is-prop-valid"));
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var Box = styled_1.default('div', {
    shouldForwardProp: function (prop) { return is_prop_valid_1.default(prop) && prop !== 'width' && prop !== 'height'; },
})({
    backgroundColor: canvas_kit_react_core_1.colors.frenchVanilla100,
    border: "1px solid " + canvas_kit_react_core_1.colors.soap500,
    borderRadius: canvas_kit_react_core_1.borderRadius.l,
    boxSizing: 'border-box',
}, function (_a) {
    var depth = _a.depth;
    return depth;
}, function (_a) {
    var padding = _a.padding;
    return ({ padding: padding });
}, function (_a) {
    var width = _a.width;
    return width && { width: width };
}, function (_a) {
    var height = _a.height;
    return height && { height: height };
});
var Header = styled_1.default('h3')(canvas_kit_react_core_1.type.h3, {
    marginBottom: canvas_kit_react_core_1.spacing.m,
    marginTop: 0,
});
var Body = styled_1.default('div')(canvas_kit_react_core_1.type.body);
var Card = (function (_super) {
    __extends(Card, _super);
    function Card() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Card.prototype.render = function () {
        var _a = this.props, _b = _a.depth, depth = _b === void 0 ? canvas_kit_react_core_1.depth[2] : _b, _c = _a.padding, padding = _c === void 0 ? canvas_kit_react_core_1.spacing.l : _c, heading = _a.heading, headingId = _a.headingId, elemProps = __rest(_a, ["depth", "padding", "heading", "headingId"]);
        return (React.createElement(Box, __assign({}, elemProps, { depth: depth, padding: padding }),
            heading && React.createElement(Header, { id: headingId }, heading),
            React.createElement(Body, null, this.props.children)));
    };
    return Card;
}(React.Component));
exports.default = Card;
