// @workday/canvas-kit-react-core/dist/commonjs/lib/colors.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var BrandingColor;
(function (BrandingColor) {
    BrandingColor["Cinnamon"] = "cinnamon";
    BrandingColor["Peach"] = "peach";
    BrandingColor["ChiliMango"] = "chiliMango";
    BrandingColor["Cantaloupe"] = "cantaloupe";
    BrandingColor["SourLemon"] = "sourLemon";
    BrandingColor["JuicyPear"] = "juicyPear";
    BrandingColor["Kiwi"] = "kiwi";
    BrandingColor["GreenApple"] = "greenApple";
    BrandingColor["Watermelon"] = "watermelon";
    BrandingColor["Jewel"] = "jewel";
    BrandingColor["Toothpaste"] = "toothpaste";
    BrandingColor["Blueberry"] = "blueberry";
    BrandingColor["Plum"] = "plum";
    BrandingColor["BerrySmoothie"] = "berrySmoothie";
    BrandingColor["Blackberry"] = "blackberry";
    BrandingColor["IslandPunch"] = "islandPunch";
    BrandingColor["GrapeSoda"] = "grapeSoda";
    BrandingColor["Pomegranate"] = "pomegranate";
    BrandingColor["FruitPunch"] = "fruitPunch";
    BrandingColor["RootBeer"] = "rootBeer";
    BrandingColor["ToastedMarshmallow"] = "toastedMarshmallow";
    BrandingColor["Cappuccino"] = "cappuccino";
    BrandingColor["Licorice"] = "licorice";
    BrandingColor["BlackPepper"] = "blackPepper";
})(BrandingColor = exports.BrandingColor || (exports.BrandingColor = {}));
