// @workday/canvas-kit-react-core/dist/commonjs/lib/type.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var get_1 = __importDefault(require("lodash/get"));
var canvas_colors_web_1 = __importStar(require("@workday/canvas-colors-web"));
var radius_1 = require("./radius");
var inheritFont = typeof window !== 'undefined' && get_1.default(window, 'window.workday.canvas.inheritFontFamily');
exports.fontFamily = inheritFont
    ? 'inherit'
    : '"Roboto", "Helvetica Neue", "Helvetica", Arial, sans-serif';
exports.monoFontFamily = '"Roboto Mono", "Courier New", Courier, monospace';
var hierarchy = {
    dataViz1: {
        fontSize: '56px',
        lineHeight: '68px',
        fontWeight: 300,
        color: canvas_colors_web_1.typeColors.heading,
    },
    dataViz2: {
        fontSize: '34px',
        lineHeight: '40px',
        fontWeight: 300,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h1: {
        fontSize: '28px',
        lineHeight: '36px',
        fontWeight: 500,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h2: {
        fontSize: '24px',
        lineHeight: '32px',
        fontWeight: 700,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h3: {
        fontSize: '20px',
        lineHeight: '28px',
        fontWeight: 700,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h4: {
        fontSize: '16px',
        lineHeight: '24px',
        fontWeight: 700,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h5: {
        fontSize: '16px',
        lineHeight: '24px',
        fontWeight: 400,
        color: canvas_colors_web_1.typeColors.heading,
    },
    body: {
        fontSize: '14px',
        lineHeight: '20px',
        fontWeight: 400,
        color: canvas_colors_web_1.typeColors.body,
    },
    body2: {
        fontSize: '13px',
        lineHeight: '20px',
        fontWeight: 400,
        color: canvas_colors_web_1.typeColors.body,
    },
    small: {
        fontSize: '12px',
        lineHeight: '16px',
        fontWeight: 400,
        color: canvas_colors_web_1.typeColors.body,
    },
};
Object.keys(hierarchy).forEach(function (key) {
    hierarchy[key] = __assign(__assign({}, hierarchy[key]), { fontFamily: exports.fontFamily });
});
var variants = {
    label: {
        fontWeight: 500,
    },
    button: {
        fontWeight: 500,
    },
    caps: {
        fontWeight: 700,
        textTransform: 'uppercase',
    },
    hint: {
        color: canvas_colors_web_1.typeColors.hint,
    },
    error: {
        color: canvas_colors_web_1.statusColors.error,
    },
    inverse: {
        color: canvas_colors_web_1.typeColors.inverse,
    },
    mono: {
        fontFamily: exports.monoFontFamily,
    },
    link: {
        textDecoration: 'underline',
        color: canvas_colors_web_1.default.blueberry400,
        cursor: 'pointer',
        borderRadius: radius_1.borderRadius.s,
        display: 'inline-block',
        padding: '0 2px',
        margin: '0 -2px',
        transition: 'color 0.15s,background-color 0.15s',
        '&:hover': {
            color: canvas_colors_web_1.default.blueberry500,
            background: canvas_colors_web_1.default.soap200,
        },
        '&:focus': {
            boxShadow: "0 0 0 2px " + canvas_colors_web_1.default.blueberry400,
            outline: 'none',
        },
        '&:active': {
            color: canvas_colors_web_1.default.blueberry600,
            background: canvas_colors_web_1.default.soap300,
        },
    },
};
var type = __assign(__assign({}, hierarchy), { variant: __assign({}, variants) });
exports.default = type;
