// @workday/canvas-kit-react-common/dist/commonjs/lib/theming/breakpoints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var BreakpointKey;
(function (BreakpointKey) {
    BreakpointKey["zero"] = "zero";
    BreakpointKey["s"] = "s";
    BreakpointKey["m"] = "m";
    BreakpointKey["l"] = "l";
    BreakpointKey["xl"] = "xl";
})(BreakpointKey = exports.BreakpointKey || (exports.BreakpointKey = {}));
exports.breakpointKeys = ['zero', 's', 'm', 'l', 'xl'];
exports.breakpoints = {
    zero: 0,
    s: 600,
    m: 960,
    l: 1280,
    xl: 1920,
};
var step = 0.5;
function up(key) {
    var value = typeof exports.breakpoints[key] === 'number' ? exports.breakpoints[key] : key;
    return "@media (min-width:" + value + "px)";
}
exports.up = up;
function down(key) {
    var endIndex = exports.breakpointKeys.indexOf(key) + 1;
    var upperbound = exports.breakpoints[exports.breakpointKeys[endIndex]];
    if (endIndex === exports.breakpointKeys.length) {
        return up(BreakpointKey.zero);
    }
    var value = typeof upperbound === 'number' && endIndex > 0 ? upperbound : 0;
    return "@media (max-width:" + (value - step) + "px)";
}
exports.down = down;
function between(start, end) {
    var endIndex = exports.breakpointKeys.indexOf(end) + 1;
    if (endIndex === exports.breakpointKeys.length) {
        return up(start);
    }
    return ("@media (min-width:" + exports.breakpoints[start] + "px) and " +
        ("(max-width:" + (exports.breakpoints[exports.breakpointKeys[endIndex]] - step) + "px)"));
}
exports.between = between;
function only(key) {
    return between(key, key);
}
exports.only = only;
