// @workday/canvas-kit-react-common/dist/commonjs/lib/theming/createCanvasTheme.js
"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var chroma_js_1 = __importDefault(require("chroma-js"));
var merge_1 = __importDefault(require("lodash/merge"));
var memoize_1 = __importDefault(require("lodash/memoize"));
var findKey_1 = __importDefault(require("lodash/findKey"));
var canvas_colors_web_1 = __importDefault(require("@workday/canvas-colors-web"));
var theme_1 = require("./theme");
var types_1 = require("./types");
var utils_1 = require("../utils");
var ColorDirection;
(function (ColorDirection) {
    ColorDirection[ColorDirection["Darken"] = 0] = "Darken";
    ColorDirection[ColorDirection["Brighten"] = 1] = "Brighten";
})(ColorDirection = exports.ColorDirection || (exports.ColorDirection = {}));
function shiftColor(hexColor, direction) {
    var canvasColor = findKey_1.default(canvas_colors_web_1.default, function (color) { return color === hexColor; });
    var darken = direction === ColorDirection.Darken;
    if (canvasColor) {
        var colorRegex = /([a-zAz]*)(\d{3})/g;
        var match = colorRegex.exec(canvasColor);
        if (match) {
            var baseColor = match[1];
            var shadeNumber = parseInt(match[2], 10);
            var newShade = darken ? shadeNumber + 100 : shadeNumber - 100;
            if (newShade >= 100 && newShade <= 600) {
                return canvas_colors_web_1.default[(baseColor + newShade)];
            }
        }
    }
    try {
        var newColor = darken ? chroma_js_1.default(hexColor).darken() : chroma_js_1.default(hexColor).brighten();
        return newColor.hex();
    }
    catch (e) {
        console.warn("Invalid color '" + hexColor + "' used in theme");
        return hexColor;
    }
}
exports.shiftColor = shiftColor;
function fillPalette(defaultPalette, palette) {
    if (!palette) {
        return {};
    }
    var main = palette.main || defaultPalette.main;
    var dark = palette.dark ||
        (palette.main && shiftColor(palette.main, ColorDirection.Darken)) ||
        defaultPalette.dark;
    var darkest = palette.darkest ||
        (palette.main && shiftColor(dark, ColorDirection.Darken)) ||
        defaultPalette.darkest;
    var light = palette.light ||
        (palette.main && shiftColor(palette.main, ColorDirection.Brighten)) ||
        defaultPalette.light;
    var lightest = palette.lightest ||
        (palette.main && shiftColor(light, ColorDirection.Brighten)) ||
        defaultPalette.lightest;
    var contrast = palette.contrast || utils_1.pickForegroundColor(main) || defaultPalette.contrast;
    return {
        lightest: lightest,
        light: light,
        main: main,
        dark: dark,
        darkest: darkest,
        contrast: contrast,
    };
}
function calculateCanvasTheme(partialTheme) {
    var _a = partialTheme.palette, palette = _a === void 0 ? {} : _a, _b = partialTheme.breakpoints, breakpoints = _b === void 0 ? {} : _b, direction = partialTheme.direction, extraFields = __rest(partialTheme, ["palette", "breakpoints", "direction"]);
    var _c = palette, primary = _c.primary, alert = _c.alert, error = _c.error, success = _c.success, neutral = _c.neutral, _d = _c.common, common = _d === void 0 ? {} : _d;
    var mergeable = {
        palette: {
            common: common,
            primary: fillPalette(theme_1.defaultCanvasTheme.palette.primary, primary),
            alert: fillPalette(theme_1.defaultCanvasTheme.palette.alert, alert),
            error: fillPalette(theme_1.defaultCanvasTheme.palette.error, error),
            success: fillPalette(theme_1.defaultCanvasTheme.palette.success, success),
            neutral: fillPalette(theme_1.defaultCanvasTheme.palette.neutral, neutral),
        },
        breakpoints: breakpoints,
        direction: direction === types_1.ContentDirection.RTL ? direction : types_1.ContentDirection.LTR,
    };
    return merge_1.default({}, theme_1.defaultCanvasTheme, mergeable, extraFields);
}
exports.createCanvasTheme = memoize_1.default(calculateCanvasTheme, function () {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    return JSON.stringify(args);
});
