// @workday/canvas-kit-react-common/dist/commonjs/lib/theming/theme.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var canvas_colors_web_1 = __importDefault(require("@workday/canvas-colors-web"));
var types_1 = require("./types");
var breakpoints_1 = require("./breakpoints");
exports.defaultCanvasTheme = {
    palette: {
        primary: {
            lightest: canvas_colors_web_1.default.blueberry100,
            light: canvas_colors_web_1.default.blueberry200,
            main: canvas_colors_web_1.default.blueberry400,
            dark: canvas_colors_web_1.default.blueberry500,
            darkest: canvas_colors_web_1.default.blueberry600,
            contrast: canvas_colors_web_1.default.frenchVanilla100,
        },
        alert: {
            lightest: canvas_colors_web_1.default.cantaloupe100,
            light: canvas_colors_web_1.default.cantaloupe300,
            main: canvas_colors_web_1.default.cantaloupe400,
            dark: canvas_colors_web_1.default.cantaloupe500,
            darkest: canvas_colors_web_1.default.cantaloupe600,
            contrast: canvas_colors_web_1.default.frenchVanilla100,
        },
        error: {
            lightest: canvas_colors_web_1.default.cinnamon100,
            light: canvas_colors_web_1.default.cinnamon200,
            main: canvas_colors_web_1.default.cinnamon500,
            dark: canvas_colors_web_1.default.cinnamon600,
            darkest: '#80160E',
            contrast: canvas_colors_web_1.default.frenchVanilla100,
        },
        success: {
            lightest: canvas_colors_web_1.default.greenApple100,
            light: canvas_colors_web_1.default.greenApple300,
            main: canvas_colors_web_1.default.greenApple600,
            dark: '',
            darkest: '',
            contrast: canvas_colors_web_1.default.frenchVanilla100,
        },
        neutral: {
            lightest: canvas_colors_web_1.default.soap200,
            light: canvas_colors_web_1.default.soap300,
            main: canvas_colors_web_1.default.soap600,
            dark: canvas_colors_web_1.default.licorice300,
            darkest: canvas_colors_web_1.default.licorice400,
            contrast: canvas_colors_web_1.default.frenchVanilla100,
        },
        common: {
            focusOutline: canvas_colors_web_1.default.blueberry400,
        },
    },
    breakpoints: {
        values: breakpoints_1.breakpoints,
        up: breakpoints_1.up,
        down: breakpoints_1.down,
        between: breakpoints_1.between,
        only: breakpoints_1.only,
    },
    direction: types_1.ContentDirection.LTR,
};
