// @workday/canvas-kit-react-core/dist/commonjs/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var canvasColorsWeb = __importStar(require("@workday/canvas-colors-web"));
var radius_1 = require("./lib/radius");
exports.borderRadius = radius_1.borderRadius;
var colors_types_1 = require("./lib/colors.types");
exports.BrandingColor = colors_types_1.BrandingColor;
var depth_1 = __importDefault(require("./lib/depth"));
exports.depth = depth_1.default;
var InputProvider_1 = __importDefault(require("./lib/InputProvider"));
exports.InputProvider = InputProvider_1.default;
var spacing_1 = __importStar(require("./lib/spacing"));
exports.spacing = spacing_1.default;
exports.spacingNumbers = spacing_1.spacingNumbers;
var type_1 = __importStar(require("./lib/type"));
exports.type = type_1.default;
exports.fontFamily = type_1.fontFamily;
exports.monoFontFamily = type_1.monoFontFamily;
var colors = canvasColorsWeb.default, gradients = canvasColorsWeb.gradients, semanticColors = __rest(canvasColorsWeb, ["default", "gradients"]);
exports.colors = colors;
exports.gradients = gradients;
var canvas = __assign({ colors: colors,
    depth: depth_1.default,
    spacing: spacing_1.default,
    type: type_1.default,
    fontFamily: type_1.fontFamily,
    monoFontFamily: type_1.monoFontFamily }, semanticColors);
__export(require("./lib/TypeWrappers"));
__export(require("@workday/canvas-colors-web"));
exports.default = canvas;
