// @workday/canvas-colors-web/dist/commonjs/canvas-colors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.typeColors = exports.statusColors = exports.inputColors = exports.iconColors = exports.buttonColors = exports.commonColors = exports.chartingColorOffsets = exports.chartingColors = exports.gradients = exports.colors = void 0;
exports.colors = {
    cinnamon600: 'hsla(3.7, 80.1%, 35.5%, 1)',
    cinnamon500: 'hsla(4.1, 74.1%, 50%, 1)',
    cinnamon400: 'hsla(3.9, 100%, 63.9%, 1)',
    cinnamon300: 'hsla(4.2, 100%, 74.5%, 1)',
    cinnamon200: 'hsla(4, 91.6%, 88.1%, 1)',
    cinnamon100: 'hsla(3.5, 100%, 96.7%, 1)',
    peach600: 'hsla(12.2, 81%, 39.2%, 1)',
    peach500: 'hsla(12.1, 74.1%, 50%, 1)',
    peach400: 'hsla(12.1, 100%, 62%, 1)',
    peach300: 'hsla(12.2, 100%, 73.9%, 1)',
    peach200: 'hsla(11.8, 100%, 85.1%, 1)',
    peach100: 'hsla(12, 100%, 97.1%, 1)',
    chiliMango600: 'hsla(19.9, 100%, 32%, 1)',
    chiliMango500: 'hsla(20.1, 100%, 43.9%, 1)',
    chiliMango400: 'hsla(20, 100%, 55.3%, 1)',
    chiliMango300: 'hsla(20, 100%, 70.6%, 1)',
    chiliMango200: 'hsla(20, 100%, 83.5%, 1)',
    chiliMango100: 'hsla(20, 100%, 92.5%, 1)',
    cantaloupe600: 'hsla(33.8, 100%, 37.6%, 1)',
    cantaloupe500: 'hsla(34.3, 100%, 47.6%, 1)',
    cantaloupe400: 'hsla(34, 100%, 57.5%, 1)',
    cantaloupe300: 'hsla(34.2, 100%, 69.4%, 1)',
    cantaloupe200: 'hsla(34.2, 93.9%, 80.6%, 1)',
    cantaloupe100: 'hsla(34, 100%, 92.5%, 1)',
    sourLemon600: 'hsla(46, 100%, 37%, 1)',
    sourLemon500: 'hsla(46, 100%, 46%, 1)',
    sourLemon400: 'hsla(44, 100%, 58%, 1)',
    sourLemon300: 'hsla(45.9, 100%, 69%, 1)',
    sourLemon200: 'hsla(46.4, 100%, 83.5%, 1)',
    sourLemon100: 'hsla(45.6, 100%, 95.1%, 1)',
    juicyPear600: 'hsla(70, 66.7%, 28.2%, 1)',
    juicyPear500: 'hsla(70.1, 74.7%, 37.3%, 1)',
    juicyPear400: 'hsla(69.9, 68.7%, 45.1%, 1)',
    juicyPear300: 'hsla(69.9, 70.5%, 56.1%, 1)',
    juicyPear200: 'hsla(70.4, 80.3%, 76.1%, 1)',
    juicyPear100: 'hsla(69, 66.7%, 94.1%, 1)',
    kiwi600: 'hsla(86.4, 53.8%, 30.6%, 1)',
    kiwi500: 'hsla(85.9, 75.9%, 34.1%, 1)',
    kiwi400: 'hsla(86.4, 71.7%, 42.9%, 1)',
    kiwi300: 'hsla(85.9, 68%, 62%, 1)',
    kiwi200: 'hsla(86.3, 83.1%, 76.9%, 1)',
    kiwi100: 'hsla(85.9, 86%, 91.6%, 1)',
    greenApple600: 'hsla(134.8, 57.4%, 30.4%, 1)',
    greenApple500: 'hsla(135.1, 52.2%, 40.2%, 1)',
    greenApple400: 'hsla(134.9, 52.2%, 51.6%, 1)',
    greenApple300: 'hsla(135, 70.2%, 63.1%, 1)',
    greenApple200: 'hsla(134.8, 78.5%, 81.8%, 1)',
    greenApple100: 'hsla(135, 100%, 96.1%, 1)',
    watermelon600: 'hsla(162.8, 100%, 17.1%, 1)',
    watermelon500: 'hsla(163.1, 82.1%, 26.3%, 1)',
    watermelon400: 'hsla(163, 80.4%, 36.1%, 1)',
    watermelon300: 'hsla(163.1, 50.2%, 59.8%, 1)',
    watermelon200: 'hsla(163.3, 60%, 82.4%, 1)',
    watermelon100: 'hsla(163.3, 81.8%, 95.7%, 1)',
    jewel600: 'hsla(186.4, 69.1%, 26.7%, 1)',
    jewel500: 'hsla(185.8, 68.7%, 32.5%, 1)',
    jewel400: 'hsla(186, 71.3%, 41%, 1)',
    jewel300: 'hsla(186.1, 64.8%, 55.5%, 1)',
    jewel200: 'hsla(185.9, 74.7%, 81.4%, 1)',
    jewel100: 'hsla(186, 100%, 96.1%, 1)',
    toothpaste600: 'hsla(198, 100%, 25.5%, 1)',
    toothpaste500: 'hsla(198.1, 97.5%, 32%, 1)',
    toothpaste400: 'hsla(198, 78.7%, 44.1%, 1)',
    toothpaste300: 'hsla(197.8, 76%, 57.5%, 1)',
    toothpaste200: 'hsla(198.2, 100%, 80%, 1)',
    toothpaste100: 'hsla(197.8, 86%, 91.6%, 1)',
    blueberry600: 'hsla(210.2, 100%, 26.5%, 1)',
    blueberry500: 'hsla(210.2, 100%, 36.3%, 1)',
    blueberry400: 'hsla(209.9, 93.1%, 45.7%, 1)',
    blueberry300: 'hsla(209.8, 100%, 62.5%, 1)',
    blueberry200: 'hsla(210.3, 100%, 82.5%, 1)',
    blueberry100: 'hsla(210, 88.1%, 91.6%, 1)',
    plum600: 'hsla(214.3, 52.5%, 31.4%, 1)',
    plum500: 'hsla(213.9, 55.5%, 43.1%, 1)',
    plum400: 'hsla(214.1, 73.8%, 55.1%, 1)',
    plum300: 'hsla(213.9, 94.4%, 65.1%, 1)',
    plum200: 'hsla(213.7, 100%, 82.5%, 1)',
    plum100: 'hsla(213.6, 100%, 95.1%, 1)',
    berrySmoothie600: 'hsla(228.9, 39.2%, 38%, 1)',
    berrySmoothie500: 'hsla(229, 40.9%, 49.8%, 1)',
    berrySmoothie400: 'hsla(229, 73.1%, 63.5%, 1)',
    berrySmoothie300: 'hsla(228.9, 100%, 73.5%, 1)',
    berrySmoothie200: 'hsla(227, 100%, 88%, 1)',
    berrySmoothie100: 'hsla(227, 100%, 95.5%, 1)',
    blackberry600: 'hsla(240.6, 52.6%, 37.3%, 1)',
    blackberry500: 'hsla(240.9, 58%, 52.4%, 1)',
    blackberry400: 'hsla(241.3, 73.8%, 62.5%, 1)',
    blackberry300: 'hsla(240.6, 66.4%, 70.8%, 1)',
    blackberry200: 'hsla(241, 100%, 88%, 1)',
    blackberry100: 'hsla(240, 100%, 97.1%, 1)',
    islandPunch600: 'hsla(259.5, 39.8%, 36.5%, 1)',
    islandPunch500: 'hsla(259.6, 40%, 45.1%, 1)',
    islandPunch400: 'hsla(260.2, 55.1%, 59.8%, 1)',
    islandPunch300: 'hsla(259.6, 64.8%, 72.2%, 1)',
    islandPunch200: 'hsla(260, 85.7%, 86.3%, 1)',
    islandPunch100: 'hsla(260, 100%, 97.1%, 1)',
    grapeSoda600: 'hsla(295.1, 39.8%, 36.5%, 1)',
    grapeSoda500: 'hsla(295.1, 36.8%, 45.3%, 1)',
    grapeSoda400: 'hsla(295.2, 55.1%, 59.8%, 1)',
    grapeSoda300: 'hsla(294.8, 64.8%, 72.2%, 1)',
    grapeSoda200: 'hsla(295.2, 100%, 87.6%, 1)',
    grapeSoda100: 'hsla(297, 100%, 96.1%, 1)',
    pomegranate600: 'hsla(337.3, 100%, 30%, 1)',
    pomegranate500: 'hsla(336.8, 95.1%, 40%, 1)',
    pomegranate400: 'hsla(337.2, 90.4%, 51%, 1)',
    pomegranate300: 'hsla(337.2, 100%, 68%, 1)',
    pomegranate200: 'hsla(337.3, 100%, 87.1%, 1)',
    pomegranate100: 'hsla(336, 100%, 96.1%, 1)',
    fruitPunch600: 'hsla(0, 64.3%, 43.9%, 1)',
    fruitPunch500: 'hsla(0, 74.8%, 53.3%, 1)',
    fruitPunch400: 'hsla(0, 100%, 64.9%, 1)',
    fruitPunch300: 'hsla(0, 100%, 74.7%, 1)',
    fruitPunch200: 'hsla(0, 100%, 87.1%, 1)',
    fruitPunch100: 'hsla(0, 100%, 96.7%, 1)',
    rootBeer600: 'hsla(18.3, 21.4%, 32.9%, 1)',
    rootBeer500: 'hsla(18.9, 15.7%, 47.5%, 1)',
    rootBeer400: 'hsla(18.3, 25%, 63.9%, 1)',
    rootBeer300: 'hsla(17.9, 40.2%, 77.1%, 1)',
    rootBeer200: 'hsla(17.1, 41.2%, 86.7%, 1)',
    rootBeer100: 'hsla(18, 50%, 96.1%, 1)',
    toastedMarshmallow600: 'hsla(41.1, 100%, 27.5%, 1)',
    toastedMarshmallow500: 'hsla(40.9, 83.6%, 38.2%, 1)',
    toastedMarshmallow400: 'hsla(41, 58.7%, 51.6%, 1)',
    toastedMarshmallow300: 'hsla(40.8, 70.9%, 66.3%, 1)',
    toastedMarshmallow200: 'hsla(40.9, 62.3%, 79.2%, 1)',
    toastedMarshmallow100: 'hsla(41.7, 85.2%, 94.7%, 1)',
    coconut600: 'hsla(353.3, 3.9%, 54.3%, 1)',
    coconut500: 'hsla(0, 4.4%, 60.2%, 1)',
    coconut400: 'hsla(0, 4.4%, 68.8%, 1)',
    coconut300: 'hsla(350, 6.1%, 80.8%, 1)',
    coconut200: 'hsla(0, 6.7%, 88.2%, 1)',
    coconut100: 'hsla(0, 6.2%, 93.7%, 1)',
    cappuccino600: 'hsla(345, 6.1%, 12.9%, 1)',
    cappuccino500: 'hsla(0, 6%, 19.6%, 1)',
    cappuccino400: 'hsla(0, 5.7%, 27.5%, 1)',
    cappuccino300: 'hsla(0, 3.9%, 35.5%, 1)',
    cappuccino200: 'hsla(352.5, 3.7%, 42.4%, 1)',
    cappuccino100: 'hsla(351.4, 3%, 46.5%, 1)',
    soap600: 'hsla(210, 11.1%, 75.3%, 1)',
    soap500: 'hsla(210, 12.4%, 82.9%, 1)',
    soap400: 'hsla(210, 11.9%, 88.7%, 1)',
    soap300: 'hsla(210, 11.7%, 92.1%, 1)',
    soap200: 'hsla(210, 8.7%, 94.5%, 1)',
    soap100: 'hsla(210, 12.5%, 96.9%, 1)',
    licorice600: 'hsla(210, 19.8%, 15%, 1)',
    licorice500: 'hsla(210, 16.4%, 23.9%, 1)',
    licorice400: 'hsla(210, 13.6%, 33.4%, 1)',
    licorice300: 'hsla(210, 11.1%, 41.4%, 1)',
    licorice200: 'hsla(210, 8.2%, 52.2%, 1)',
    licorice100: 'hsla(210, 10.6%, 66.7%, 1)',
    frenchVanilla600: 'hsla(0, 0%, 56.1%, 1)',
    frenchVanilla500: 'hsla(0, 0%, 65.1%, 1)',
    frenchVanilla400: 'hsla(0, 0%, 74.1%, 1)',
    frenchVanilla300: 'hsla(0, 0%, 83.1%, 1)',
    frenchVanilla200: 'hsla(0, 0%, 92.2%, 1)',
    frenchVanilla100: 'hsla(0, 0%, 100%, 1)',
    blackPepper600: 'hsla(0, 0%, 0%, 1)',
    blackPepper500: 'hsla(0, 0%, 11.8%, 1)',
    blackPepper400: 'hsla(0, 0%, 20%, 1)',
    blackPepper300: 'hsla(0, 0%, 28.6%, 1)',
    blackPepper200: 'hsla(0, 0%, 38%, 1)',
    blackPepper100: 'hsla(0, 0%, 47.1%, 1)',
};
exports.gradients = {
    cinnamon: "linear-gradient(to bottom right," + exports.colors.cinnamon500 + "," + exports.colors.cinnamon400 + ")",
    peach: "linear-gradient(to bottom right," + exports.colors.peach500 + "," + exports.colors.peach400 + ")",
    chiliMango: "linear-gradient(to bottom right," + exports.colors.chiliMango500 + "," + exports.colors.chiliMango400 + ")",
    cantaloupe: "linear-gradient(to bottom right," + exports.colors.cantaloupe500 + "," + exports.colors.cantaloupe400 + ")",
    sourLemon: "linear-gradient(to bottom right," + exports.colors.sourLemon500 + "," + exports.colors.sourLemon400 + ")",
    juicyPear: "linear-gradient(to bottom right," + exports.colors.juicyPear500 + "," + exports.colors.juicyPear400 + ")",
    kiwi: "linear-gradient(to bottom right," + exports.colors.kiwi500 + "," + exports.colors.kiwi400 + ")",
    greenApple: "linear-gradient(to bottom right," + exports.colors.greenApple500 + "," + exports.colors.greenApple400 + ")",
    watermelon: "linear-gradient(to bottom right," + exports.colors.watermelon500 + "," + exports.colors.watermelon400 + ")",
    jewel: "linear-gradient(to bottom right," + exports.colors.jewel500 + "," + exports.colors.jewel400 + ")",
    toothpaste: "linear-gradient(to bottom right," + exports.colors.toothpaste500 + "," + exports.colors.toothpaste400 + ")",
    blueberry: "linear-gradient(to bottom right," + exports.colors.blueberry500 + "," + exports.colors.blueberry400 + ")",
    plum: "linear-gradient(to bottom right," + exports.colors.plum500 + "," + exports.colors.plum400 + ")",
    berrySmoothie: "linear-gradient(to bottom right," + exports.colors.berrySmoothie500 + "," + exports.colors.berrySmoothie400 + ")",
    blackberry: "linear-gradient(to bottom right," + exports.colors.blackberry500 + "," + exports.colors.blackberry400 + ")",
    islandPunch: "linear-gradient(to bottom right," + exports.colors.islandPunch500 + "," + exports.colors.islandPunch400 + ")",
    grapeSoda: "linear-gradient(to bottom right," + exports.colors.grapeSoda500 + "," + exports.colors.grapeSoda400 + ")",
    pomegranate: "linear-gradient(to bottom right," + exports.colors.pomegranate500 + "," + exports.colors.pomegranate400 + ")",
    fruitPunch: "linear-gradient(to bottom right," + exports.colors.fruitPunch500 + "," + exports.colors.fruitPunch400 + ")",
    rootBeer: "linear-gradient(to bottom right," + exports.colors.rootBeer600 + "," + exports.colors.rootBeer500 + ")",
    toastedMarshmallow: "linear-gradient(to bottom right," + exports.colors.toastedMarshmallow500 + "," + exports.colors.toastedMarshmallow400 + ")",
    cappuccino: "linear-gradient(to bottom right," + exports.colors.cappuccino400 + "," + exports.colors.cappuccino300 + ")",
    licorice: "linear-gradient(to bottom right," + exports.colors.licorice500 + "," + exports.colors.licorice400 + ")",
    blackPepper: "linear-gradient(to bottom right," + exports.colors.blackPepper400 + "," + exports.colors.blackPepper300 + ")",
};
exports.default = exports.colors;
exports.chartingColors = {
    1: exports.colors.watermelon200,
    2: exports.colors.watermelon300,
    3: exports.colors.jewel200,
    4: exports.colors.jewel300,
    5: exports.colors.plum200,
    6: exports.colors.plum300,
    7: exports.colors.islandPunch200,
    8: exports.colors.islandPunch300,
    9: exports.colors.pomegranate200,
    10: exports.colors.pomegranate300,
    11: exports.colors.peach200,
    12: exports.colors.peach400,
    13: exports.colors.cantaloupe200,
    14: exports.colors.cantaloupe400,
    15: exports.colors.sourLemon200,
    16: exports.colors.sourLemon400,
    17: exports.colors.watermelon400,
    18: exports.colors.watermelon500,
    19: exports.colors.jewel400,
    20: exports.colors.jewel500,
    21: exports.colors.plum400,
    22: exports.colors.plum500,
    23: exports.colors.islandPunch400,
    24: exports.colors.islandPunch500,
    25: exports.colors.pomegranate400,
    26: exports.colors.pomegranate600,
    27: exports.colors.peach500,
    28: exports.colors.peach600,
    29: exports.colors.cantaloupe500,
    30: exports.colors.cantaloupe600,
    31: exports.colors.sourLemon500,
    32: exports.colors.sourLemon600,
    33: exports.colors.greenApple200,
    34: exports.colors.greenApple300,
    35: exports.colors.toothpaste200,
    36: exports.colors.toothpaste300,
    37: exports.colors.blueberry200,
    38: exports.colors.blueberry300,
    39: exports.colors.berrySmoothie200,
    40: exports.colors.blackberry300,
    41: exports.colors.fruitPunch200,
    42: exports.colors.fruitPunch300,
    43: exports.colors.rootBeer200,
    44: exports.colors.rootBeer300,
    45: exports.colors.toastedMarshmallow200,
    46: exports.colors.toastedMarshmallow400,
    47: exports.colors.chiliMango200,
    48: exports.colors.chiliMango400,
    49: exports.colors.greenApple400,
    50: exports.colors.greenApple600,
    51: exports.colors.toothpaste400,
    52: exports.colors.toothpaste600,
    53: exports.colors.blueberry400,
    54: exports.colors.blueberry600,
    55: exports.colors.blackberry400,
    56: exports.colors.blackberry600,
    57: exports.colors.fruitPunch400,
    58: exports.colors.fruitPunch600,
    59: exports.colors.rootBeer500,
    60: exports.colors.rootBeer600,
    61: exports.colors.toastedMarshmallow500,
    62: exports.colors.toastedMarshmallow600,
    63: exports.colors.chiliMango500,
    64: exports.colors.chiliMango600,
    65: exports.colors.kiwi200,
    66: exports.colors.kiwi400,
    67: exports.colors.berrySmoothie200,
    68: exports.colors.berrySmoothie300,
    69: exports.colors.grapeSoda200,
    70: exports.colors.grapeSoda300,
    71: exports.colors.cinnamon200,
    72: exports.colors.cinnamon300,
    73: exports.colors.sourLemon300,
    74: exports.colors.toastedMarshmallow300,
    75: exports.colors.peach300,
    76: exports.colors.cinnamon500,
    77: exports.colors.cantaloupe300,
    78: exports.colors.chiliMango300,
    79: exports.colors.kiwi300,
    80: exports.colors.kiwi500,
    81: exports.colors.berrySmoothie400,
    82: exports.colors.berrySmoothie500,
    83: exports.colors.grapeSoda400,
    84: exports.colors.grapeSoda500,
    85: exports.colors.cinnamon400,
    86: exports.colors.cinnamon500,
};
exports.chartingColorOffsets = {
    barAndColumn: 1,
    pie: 4,
    lineAndArea: 7,
    bubble: 11,
};
exports.commonColors = {
    background: exports.colors.frenchVanilla100,
    backgroundAlt: exports.colors.soap200,
    focusOutline: exports.colors.blueberry400,
    focusBackground: exports.colors.blueberry400,
    hoverBackground: exports.colors.soap300,
    divider: exports.colors.soap400,
};
exports.buttonColors = {
    delete: {
        background: exports.colors.cinnamon100,
        border: exports.colors.cinnamon400,
        text: exports.colors.cinnamon500,
        activeBackground: exports.colors.cinnamon100,
        activeBorder: exports.colors.cinnamon400,
        activeText: exports.colors.cinnamon500,
        disabledBackground: exports.colors.cinnamon100,
        disabledBorder: exports.colors.cinnamon200,
        disabledText: exports.colors.cinnamon300,
        focusBackground: exports.colors.cinnamon100,
        focusBorder: exports.colors.blueberry400,
        focusText: exports.colors.cinnamon500,
        hoverBackground: exports.colors.cinnamon400,
        hoverBorder: exports.colors.cinnamon600,
        hoverText: exports.colors.frenchVanilla100,
    },
    primary: {
        background: exports.colors.cantaloupe400,
        border: exports.colors.cantaloupe400,
        text: exports.colors.frenchVanilla100,
        activeBackground: exports.colors.cantaloupe400,
        activeBorder: exports.colors.cantaloupe400,
        activeText: exports.colors.frenchVanilla100,
        disabledBackground: exports.colors.cantaloupe200,
        disabledBorder: exports.colors.cantaloupe400,
        disabledText: exports.colors.frenchVanilla100,
        focusBackground: exports.colors.cantaloupe400,
        focusBorder: exports.colors.blueberry400,
        focusText: exports.colors.frenchVanilla100,
        hoverBackground: exports.colors.cantaloupe500,
        hoverBorder: exports.colors.cantaloupe400,
        hoverText: exports.colors.frenchVanilla100,
    },
    secondary: {
        background: exports.colors.soap200,
        border: exports.colors.soap500,
        text: exports.colors.blackPepper400,
        activeBackground: exports.colors.soap200,
        activeBorder: exports.colors.soap500,
        activeText: exports.colors.blackPepper400,
        disabledBackground: exports.colors.frenchVanilla100,
        disabledBorder: exports.colors.soap500,
        disabledText: exports.colors.soap500,
        focusBackground: exports.colors.soap200,
        focusBorder: exports.colors.blueberry400,
        focusText: exports.colors.blackPepper400,
        hoverBackground: exports.colors.soap500,
        hoverBorder: exports.colors.licorice100,
        hoverText: exports.colors.blackPepper400,
    },
};
exports.iconColors = {
    active: exports.colors.blueberry500,
    disabled: exports.colors.soap600,
    hover: exports.colors.licorice500,
    inverse: exports.colors.frenchVanilla100,
    standard: exports.colors.licorice200,
};
exports.inputColors = {
    background: exports.colors.frenchVanilla100,
    border: exports.colors.licorice200,
    placeholder: exports.colors.licorice300,
    text: exports.colors.blackPepper300,
    icon: exports.colors.licorice200,
    iconHover: exports.colors.licorice500,
    selectionControlFill: exports.colors.blueberry400,
    hoverBorder: exports.colors.licorice500,
    focusBorder: exports.colors.blueberry400,
    disabled: {
        background: exports.colors.soap100,
        border: exports.colors.soap600,
        text: exports.colors.licorice100,
        icon: exports.colors.soap600,
    },
    error: {
        border: exports.colors.cinnamon500,
        message: exports.colors.blackPepper100,
        icon: exports.colors.cinnamon500,
    },
    alert: {
        border: exports.colors.cantaloupe400,
        message: exports.colors.blackPepper100,
        icon: exports.colors.cantaloupe500,
    },
};
exports.statusColors = {
    active: exports.colors.blueberry500,
    success: exports.colors.greenApple400,
    warning: exports.colors.cantaloupe400,
    error: exports.colors.cinnamon500,
};
exports.typeColors = {
    body: exports.colors.blackPepper300,
    heading: exports.colors.blackPepper400,
    hint: exports.colors.licorice300,
    inverse: exports.colors.frenchVanilla100,
    label: exports.colors.blackPepper100,
    link: exports.colors.blueberry400,
    selectHighlight: exports.colors.blueberry200,
    selectHighlightInverse: exports.colors.frenchVanilla100,
};
