// @workday/canvas-colors-web/dist/commonjs/canvas-colors-hex.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.typeColors = exports.statusColors = exports.inputColors = exports.iconColors = exports.buttonColors = exports.commonColors = exports.chartingColorOffsets = exports.chartingColors = exports.gradients = exports.colors = void 0;
exports.colors = {
    cinnamon600: '#a31b12',
    cinnamon500: '#de2e21',
    cinnamon400: '#ff5347',
    cinnamon300: '#ff867d',
    cinnamon200: '#fcc9c5',
    cinnamon100: '#ffefee',
    peach600: '#b53413',
    peach500: '#de4721',
    peach400: '#ff643d',
    peach300: '#ff957a',
    peach200: '#ffc2b3',
    peach100: '#fff3f0',
    chiliMango600: '#a33600',
    chiliMango500: '#e04b00',
    chiliMango400: '#ff671b',
    chiliMango300: '#ff9b69',
    chiliMango200: '#ffc7ab',
    chiliMango100: '#ffe6d9',
    cantaloupe600: '#c06c00',
    cantaloupe500: '#f38b00',
    cantaloupe400: '#ffa126',
    cantaloupe300: '#ffbc63',
    cantaloupe200: '#fcd49f',
    cantaloupe100: '#ffeed9',
    sourLemon600: '#bd9100',
    sourLemon500: '#ebb400',
    sourLemon400: '#ffc629',
    sourLemon300: '#ffda61',
    sourLemon200: '#ffecab',
    sourLemon100: '#fff9e6',
    juicyPear600: '#687818',
    juicyPear500: '#8ea618',
    juicyPear400: '#a8c224',
    juicyPear300: '#c4de40',
    juicyPear200: '#e2f391',
    juicyPear100: '#f7fae6',
    kiwi600: '#537824',
    kiwi500: '#609915',
    kiwi400: '#77bc1f',
    kiwi300: '#a7e05c',
    kiwi200: '#caf593',
    kiwi100: '#ecfcd7',
    greenApple600: '#217a37',
    greenApple500: '#319c4c',
    greenApple400: '#43c463',
    greenApple300: '#5fe380',
    greenApple200: '#acf5be',
    greenApple100: '#ebfff0',
    watermelon600: '#00573e',
    watermelon500: '#0c7a5b',
    watermelon400: '#12a67c',
    watermelon300: '#65ccaf',
    watermelon200: '#b7edde',
    watermelon100: '#ebfdf8',
    jewel600: '#156973',
    jewel500: '#1a818c',
    jewel400: '#1ea4b3',
    jewel300: '#44c8d7',
    jewel200: '#acecf3',
    jewel100: '#ebfdff',
    toothpaste600: '#005b82',
    toothpaste500: '#0271a1',
    toothpaste400: '#1894c9',
    toothpaste300: '#40b4e5',
    toothpaste200: '#99e0ff',
    toothpaste100: '#d7f1fc',
    blueberry600: '#004387',
    blueberry500: '#005cb9',
    blueberry400: '#0875e1',
    blueberry300: '#40a0ff',
    blueberry200: '#a6d2ff',
    blueberry100: '#d7eafc',
    plum600: '#264a7a',
    plum500: '#3166ab',
    plum400: '#3881e1',
    plum300: '#529bfa',
    plum200: '#a6cdff',
    plum100: '#e6f1ff',
    berrySmoothie600: '#3b4987',
    berrySmoothie500: '#4b5eb3',
    berrySmoothie400: '#5e77e6',
    berrySmoothie300: '#7891ff',
    berrySmoothie200: '#c2cfff',
    berrySmoothie100: '#e8edff',
    blackberry600: '#2e2d91',
    blackberry500: '#413fcc',
    blackberry400: '#5c59e6',
    blackberry300: '#8483e6',
    blackberry200: '#c3c2ff',
    blackberry100: '#f0f0ff',
    islandPunch600: '#503882',
    islandPunch500: '#6345a1',
    islandPunch400: '#8660d1',
    islandPunch300: '#a88ae6',
    islandPunch200: '#d2befa',
    islandPunch100: '#f5f0ff',
    grapeSoda600: '#7c3882',
    grapeSoda500: '#97499e',
    grapeSoda400: '#c860d1',
    grapeSoda300: '#de8ae6',
    grapeSoda200: '#fac0ff',
    grapeSoda100: '#feebff',
    pomegranate600: '#99003a',
    pomegranate500: '#c70550',
    pomegranate400: '#f31167',
    pomegranate300: '#ff5c9a',
    pomegranate200: '#ffbdd6',
    pomegranate100: '#ffebf3',
    fruitPunch600: '#b82828',
    fruitPunch500: '#e12f2f',
    fruitPunch400: '#ff4c4c',
    fruitPunch300: '#ff7e7e',
    fruitPunch200: '#ffbdbd',
    fruitPunch100: '#ffeeee',
    rootBeer600: '#664d42',
    rootBeer500: '#8c7266',
    rootBeer400: '#ba9a8c',
    rootBeer300: '#dcbbad',
    rootBeer200: '#ebd7cf',
    rootBeer100: '#faf3f0',
    toastedMarshmallow600: '#8c6000',
    toastedMarshmallow500: '#b37f10',
    toastedMarshmallow400: '#cc9e3b',
    toastedMarshmallow300: '#e6bf6c',
    toastedMarshmallow200: '#ebd6a9',
    toastedMarshmallow100: '#fdf6e6',
    coconut600: '#8f8687',
    coconut500: '#9e9595',
    coconut400: '#b3acac',
    coconut300: '#d1cbcc',
    coconut200: '#e3dfdf',
    coconut100: '#f0eeee',
    cappuccino600: '#231f20',
    cappuccino500: '#352f2f',
    cappuccino400: '#4a4242',
    cappuccino300: '#5e5757',
    cappuccino200: '#706869',
    cappuccino100: '#7a7374',
    soap600: '#b9c0c7',
    soap500: '#ced3d9',
    soap400: '#dfe2e6',
    soap300: '#e8ebed',
    soap200: '#f0f1f2',
    soap100: '#f6f7f8',
    licorice600: '#1f262e',
    licorice500: '#333d47',
    licorice400: '#4a5561',
    licorice300: '#5e6a75',
    licorice200: '#7b858f',
    licorice100: '#a1aab3',
    frenchVanilla600: '#8f8f8f',
    frenchVanilla500: '#a6a6a6',
    frenchVanilla400: '#bdbdbd',
    frenchVanilla300: '#d4d4d4',
    frenchVanilla200: '#ebebeb',
    frenchVanilla100: '#ffffff',
    blackPepper600: '#000000',
    blackPepper500: '#1e1e1e',
    blackPepper400: '#333333',
    blackPepper300: '#494949',
    blackPepper200: '#616161',
    blackPepper100: '#787878',
};
exports.gradients = {
    cinnamon: "linear-gradient(to bottom right," + exports.colors.cinnamon500 + "," + exports.colors.cinnamon400 + ")",
    peach: "linear-gradient(to bottom right," + exports.colors.peach500 + "," + exports.colors.peach400 + ")",
    chiliMango: "linear-gradient(to bottom right," + exports.colors.chiliMango500 + "," + exports.colors.chiliMango400 + ")",
    cantaloupe: "linear-gradient(to bottom right," + exports.colors.cantaloupe500 + "," + exports.colors.cantaloupe400 + ")",
    sourLemon: "linear-gradient(to bottom right," + exports.colors.sourLemon500 + "," + exports.colors.sourLemon400 + ")",
    juicyPear: "linear-gradient(to bottom right," + exports.colors.juicyPear500 + "," + exports.colors.juicyPear400 + ")",
    kiwi: "linear-gradient(to bottom right," + exports.colors.kiwi500 + "," + exports.colors.kiwi400 + ")",
    greenApple: "linear-gradient(to bottom right," + exports.colors.greenApple500 + "," + exports.colors.greenApple400 + ")",
    watermelon: "linear-gradient(to bottom right," + exports.colors.watermelon500 + "," + exports.colors.watermelon400 + ")",
    jewel: "linear-gradient(to bottom right," + exports.colors.jewel500 + "," + exports.colors.jewel400 + ")",
    toothpaste: "linear-gradient(to bottom right," + exports.colors.toothpaste500 + "," + exports.colors.toothpaste400 + ")",
    blueberry: "linear-gradient(to bottom right," + exports.colors.blueberry500 + "," + exports.colors.blueberry400 + ")",
    plum: "linear-gradient(to bottom right," + exports.colors.plum500 + "," + exports.colors.plum400 + ")",
    berrySmoothie: "linear-gradient(to bottom right," + exports.colors.berrySmoothie500 + "," + exports.colors.berrySmoothie400 + ")",
    blackberry: "linear-gradient(to bottom right," + exports.colors.blackberry500 + "," + exports.colors.blackberry400 + ")",
    islandPunch: "linear-gradient(to bottom right," + exports.colors.islandPunch500 + "," + exports.colors.islandPunch400 + ")",
    grapeSoda: "linear-gradient(to bottom right," + exports.colors.grapeSoda500 + "," + exports.colors.grapeSoda400 + ")",
    pomegranate: "linear-gradient(to bottom right," + exports.colors.pomegranate500 + "," + exports.colors.pomegranate400 + ")",
    fruitPunch: "linear-gradient(to bottom right," + exports.colors.fruitPunch500 + "," + exports.colors.fruitPunch400 + ")",
    rootBeer: "linear-gradient(to bottom right," + exports.colors.rootBeer600 + "," + exports.colors.rootBeer500 + ")",
    toastedMarshmallow: "linear-gradient(to bottom right," + exports.colors.toastedMarshmallow500 + "," + exports.colors.toastedMarshmallow400 + ")",
    cappuccino: "linear-gradient(to bottom right," + exports.colors.cappuccino400 + "," + exports.colors.cappuccino300 + ")",
    licorice: "linear-gradient(to bottom right," + exports.colors.licorice500 + "," + exports.colors.licorice400 + ")",
    blackPepper: "linear-gradient(to bottom right," + exports.colors.blackPepper400 + "," + exports.colors.blackPepper300 + ")",
};
exports.default = exports.colors;
exports.chartingColors = {
    1: exports.colors.watermelon200,
    2: exports.colors.watermelon300,
    3: exports.colors.jewel200,
    4: exports.colors.jewel300,
    5: exports.colors.plum200,
    6: exports.colors.plum300,
    7: exports.colors.islandPunch200,
    8: exports.colors.islandPunch300,
    9: exports.colors.pomegranate200,
    10: exports.colors.pomegranate300,
    11: exports.colors.peach200,
    12: exports.colors.peach400,
    13: exports.colors.cantaloupe200,
    14: exports.colors.cantaloupe400,
    15: exports.colors.sourLemon200,
    16: exports.colors.sourLemon400,
    17: exports.colors.watermelon400,
    18: exports.colors.watermelon500,
    19: exports.colors.jewel400,
    20: exports.colors.jewel500,
    21: exports.colors.plum400,
    22: exports.colors.plum500,
    23: exports.colors.islandPunch400,
    24: exports.colors.islandPunch500,
    25: exports.colors.pomegranate400,
    26: exports.colors.pomegranate600,
    27: exports.colors.peach500,
    28: exports.colors.peach600,
    29: exports.colors.cantaloupe500,
    30: exports.colors.cantaloupe600,
    31: exports.colors.sourLemon500,
    32: exports.colors.sourLemon600,
    33: exports.colors.greenApple200,
    34: exports.colors.greenApple300,
    35: exports.colors.toothpaste200,
    36: exports.colors.toothpaste300,
    37: exports.colors.blueberry200,
    38: exports.colors.blueberry300,
    39: exports.colors.berrySmoothie200,
    40: exports.colors.blackberry300,
    41: exports.colors.fruitPunch200,
    42: exports.colors.fruitPunch300,
    43: exports.colors.rootBeer200,
    44: exports.colors.rootBeer300,
    45: exports.colors.toastedMarshmallow200,
    46: exports.colors.toastedMarshmallow400,
    47: exports.colors.chiliMango200,
    48: exports.colors.chiliMango400,
    49: exports.colors.greenApple400,
    50: exports.colors.greenApple600,
    51: exports.colors.toothpaste400,
    52: exports.colors.toothpaste600,
    53: exports.colors.blueberry400,
    54: exports.colors.blueberry600,
    55: exports.colors.blackberry400,
    56: exports.colors.blackberry600,
    57: exports.colors.fruitPunch400,
    58: exports.colors.fruitPunch600,
    59: exports.colors.rootBeer500,
    60: exports.colors.rootBeer600,
    61: exports.colors.toastedMarshmallow500,
    62: exports.colors.toastedMarshmallow600,
    63: exports.colors.chiliMango500,
    64: exports.colors.chiliMango600,
    65: exports.colors.kiwi200,
    66: exports.colors.kiwi400,
    67: exports.colors.berrySmoothie200,
    68: exports.colors.berrySmoothie300,
    69: exports.colors.grapeSoda200,
    70: exports.colors.grapeSoda300,
    71: exports.colors.cinnamon200,
    72: exports.colors.cinnamon300,
    73: exports.colors.sourLemon300,
    74: exports.colors.toastedMarshmallow300,
    75: exports.colors.peach300,
    76: exports.colors.cinnamon500,
    77: exports.colors.cantaloupe300,
    78: exports.colors.chiliMango300,
    79: exports.colors.kiwi300,
    80: exports.colors.kiwi500,
    81: exports.colors.berrySmoothie400,
    82: exports.colors.berrySmoothie500,
    83: exports.colors.grapeSoda400,
    84: exports.colors.grapeSoda500,
    85: exports.colors.cinnamon400,
    86: exports.colors.cinnamon500,
};
exports.chartingColorOffsets = {
    barAndColumn: 1,
    pie: 4,
    lineAndArea: 7,
    bubble: 11,
};
exports.commonColors = {
    background: exports.colors.frenchVanilla100,
    backgroundAlt: exports.colors.soap200,
    focusOutline: exports.colors.blueberry400,
    focusBackground: exports.colors.blueberry400,
    hoverBackground: exports.colors.soap300,
    divider: exports.colors.soap400,
};
exports.buttonColors = {
    delete: {
        background: exports.colors.cinnamon100,
        border: exports.colors.cinnamon400,
        text: exports.colors.cinnamon500,
        activeBackground: exports.colors.cinnamon100,
        activeBorder: exports.colors.cinnamon400,
        activeText: exports.colors.cinnamon500,
        disabledBackground: exports.colors.cinnamon100,
        disabledBorder: exports.colors.cinnamon200,
        disabledText: exports.colors.cinnamon300,
        focusBackground: exports.colors.cinnamon100,
        focusBorder: exports.colors.blueberry400,
        focusText: exports.colors.cinnamon500,
        hoverBackground: exports.colors.cinnamon400,
        hoverBorder: exports.colors.cinnamon600,
        hoverText: exports.colors.frenchVanilla100,
    },
    primary: {
        background: exports.colors.cantaloupe400,
        border: exports.colors.cantaloupe400,
        text: exports.colors.frenchVanilla100,
        activeBackground: exports.colors.cantaloupe400,
        activeBorder: exports.colors.cantaloupe400,
        activeText: exports.colors.frenchVanilla100,
        disabledBackground: exports.colors.cantaloupe200,
        disabledBorder: exports.colors.cantaloupe400,
        disabledText: exports.colors.frenchVanilla100,
        focusBackground: exports.colors.cantaloupe400,
        focusBorder: exports.colors.blueberry400,
        focusText: exports.colors.frenchVanilla100,
        hoverBackground: exports.colors.cantaloupe500,
        hoverBorder: exports.colors.cantaloupe400,
        hoverText: exports.colors.frenchVanilla100,
    },
    secondary: {
        background: exports.colors.soap200,
        border: exports.colors.soap500,
        text: exports.colors.blackPepper400,
        activeBackground: exports.colors.soap200,
        activeBorder: exports.colors.soap500,
        activeText: exports.colors.blackPepper400,
        disabledBackground: exports.colors.frenchVanilla100,
        disabledBorder: exports.colors.soap500,
        disabledText: exports.colors.soap500,
        focusBackground: exports.colors.soap200,
        focusBorder: exports.colors.blueberry400,
        focusText: exports.colors.blackPepper400,
        hoverBackground: exports.colors.soap500,
        hoverBorder: exports.colors.licorice100,
        hoverText: exports.colors.blackPepper400,
    },
};
exports.iconColors = {
    active: exports.colors.blueberry500,
    disabled: exports.colors.soap600,
    hover: exports.colors.licorice500,
    inverse: exports.colors.frenchVanilla100,
    standard: exports.colors.licorice200,
};
exports.inputColors = {
    background: exports.colors.frenchVanilla100,
    border: exports.colors.licorice200,
    placeholder: exports.colors.licorice300,
    text: exports.colors.blackPepper300,
    icon: exports.colors.licorice200,
    iconHover: exports.colors.licorice500,
    selectionControlFill: exports.colors.blueberry400,
    hoverBorder: exports.colors.licorice500,
    focusBorder: exports.colors.blueberry400,
    disabled: {
        background: exports.colors.soap100,
        border: exports.colors.soap600,
        text: exports.colors.licorice100,
        icon: exports.colors.soap600,
    },
    error: {
        border: exports.colors.cinnamon500,
        message: exports.colors.blackPepper100,
        icon: exports.colors.cinnamon500,
    },
    alert: {
        border: exports.colors.cantaloupe400,
        message: exports.colors.blackPepper100,
        icon: exports.colors.cantaloupe500,
    },
};
exports.statusColors = {
    active: exports.colors.blueberry500,
    success: exports.colors.greenApple400,
    warning: exports.colors.cantaloupe400,
    error: exports.colors.cinnamon500,
};
exports.typeColors = {
    body: exports.colors.blackPepper300,
    heading: exports.colors.blackPepper400,
    hint: exports.colors.licorice300,
    inverse: exports.colors.frenchVanilla100,
    label: exports.colors.blackPepper100,
    link: exports.colors.blueberry400,
    selectHighlight: exports.colors.blueberry200,
    selectHighlightInverse: exports.colors.frenchVanilla100,
};
