// @workday/canvas-kit-react-popup/dist/es6/lib/Popup.js
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import styled from '@emotion/styled';
import { keyframes } from '@emotion/core';
import isPropValid from '@emotion/is-prop-valid';
import uuid from 'uuid/v4';
import Card from '@workday/canvas-kit-react-card';
import { IconButton } from '@workday/canvas-kit-react-button';
import { depth as depthValues, spacing } from '@workday/canvas-kit-react-core';
import { getTranslateFromOrigin, } from '@workday/canvas-kit-react-common';
import { xIcon } from '@workday/canvas-system-icons-web';
export var PopupPadding;
(function (PopupPadding) {
    PopupPadding["zero"] = "0px";
    PopupPadding["s"] = "16px";
    PopupPadding["l"] = "32px";
})(PopupPadding || (PopupPadding = {}));
var closeIconSpacing = spacing.xs;
var closeIconSpacingSmall = 10;
var popupAnimation = function (transformOrigin) {
    var translate = getTranslateFromOrigin(transformOrigin, spacing.xxs);
    return keyframes(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    0% {\n      opacity: 0;\n      transform: translate(", "px, ", "px);\n    }\n    100% {\n      opacity: 1;\n      transform: translate(0);\n    }\n  "], ["\n    0% {\n      opacity: 0;\n      transform: translate(", "px, ", "px);\n    }\n    100% {\n      opacity: 1;\n      transform: translate(0);\n    }\n  "])), translate.x, translate.y);
};
var Container = styled('div', {
    shouldForwardProp: function (prop) { return isPropValid(prop) && prop !== 'width'; },
})({
    position: 'relative',
    maxWidth: "calc(100vw - " + spacing.l + ")",
}, function (_a) {
    var width = _a.width;
    return width && { width: width };
}, function (_a) {
    var transformOrigin = _a.transformOrigin;
    if (transformOrigin === null) {
        return {};
    }
    return {
        animation: popupAnimation(transformOrigin),
        animationDuration: '150ms',
        animationTimingFunction: 'ease-out',
        transformOrigin: transformOrigin.vertical + " " + transformOrigin.horizontal,
        '.wd-no-animation &': {
            animation: 'none',
        },
    };
});
var getHeadingId = function (heading, id) { return (heading ? id : undefined); };
var getAriaLabel = function (ariaLabel, headingId) {
    var _a, _b;
    if (ariaLabel) {
        return _a = {}, _a['aria-label'] = ariaLabel, _a;
    }
    if (headingId) {
        return _b = {}, _b['aria-labelledby'] = headingId, _b;
    }
    return undefined;
};
var CloseIconContainer = styled('div')({
    position: 'absolute',
}, function (_a) {
    var closeIconSize = _a.closeIconSize;
    return ({
        right: closeIconSize === IconButton.Size.Small ? closeIconSpacingSmall : closeIconSpacing,
        top: closeIconSize === IconButton.Size.Small ? closeIconSpacingSmall : closeIconSpacing,
    });
});
var Popup = (function (_super) {
    __extends(Popup, _super);
    function Popup() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.id = uuid();
        _this.closeButtonRef = React.createRef();
        return _this;
    }
    Popup.prototype.render = function () {
        var _a = this.props, _b = _a.padding, padding = _b === void 0 ? Popup.Padding.l : _b, _c = _a.closeIconSize, closeIconSize = _c === void 0 ? 'medium' : _c, _d = _a.closeButtonAriaLabel, closeButtonAriaLabel = _d === void 0 ? 'Close' : _d, _e = _a.transformOrigin, transformOrigin = _e === void 0 ? {
            horizontal: 'center',
            vertical: 'top',
        } : _e, _f = _a.depth, depth = _f === void 0 ? depthValues[2] : _f, handleClose = _a.handleClose, width = _a.width, heading = _a.heading, popupRef = _a.popupRef, ariaLabel = _a.ariaLabel, elemProps = __rest(_a, ["padding", "closeIconSize", "closeButtonAriaLabel", "transformOrigin", "depth", "handleClose", "width", "heading", "popupRef", "ariaLabel"]);
        var headingId = getHeadingId(heading, this.id);
        return (React.createElement(Container, __assign({ transformOrigin: transformOrigin, width: width, role: "dialog", ref: popupRef }, getAriaLabel(ariaLabel, headingId), elemProps),
            handleClose && (React.createElement(CloseIconContainer, { closeIconSize: closeIconSize },
                React.createElement(IconButton, { "data-close": "close", buttonRef: this.closeButtonRef, variant: IconButton.Variant.Plain, size: closeIconSize, onClick: handleClose, icon: xIcon, "aria-label": closeButtonAriaLabel }))),
            React.createElement(Card, { depth: depth, heading: heading, headingId: headingId, width: "100%", padding: padding }, this.props.children)));
    };
    Popup.Padding = PopupPadding;
    return Popup;
}(React.Component));
export default Popup;
export var usePopup = function () {
    var _a = React.useState(false), open = _a[0], setOpen = _a[1];
    var _b = React.useState(null), anchorElement = _b[0], setAnchorElement = _b[1];
    var stackRef = React.useRef(null);
    var togglePopup = function (event) {
        setAnchorElement(event.currentTarget);
        setOpen(!open);
    };
    return {
        targetProps: {
            onClick: togglePopup,
        },
        closePopup: function () {
            setOpen(false);
        },
        popperProps: {
            open: open,
            anchorElement: anchorElement,
            ref: stackRef,
        },
        stackRef: stackRef,
    };
};
var templateObject_1;
