// @workday/canvas-kit-react-common/dist/es6/lib/styles/errorRing.js
import { ErrorType } from '../types';
import { colors, inputColors } from '@workday/canvas-kit-react-core';
import chroma from 'chroma-js';
var isAccessible = function (foreground, background) {
    if (background === void 0) { background = colors.frenchVanilla100; }
    return chroma.contrast(foreground, background) >= 3;
};
export function getErrorColors(error, theme) {
    if (error === ErrorType.Error) {
        if (theme) {
            var palette = theme.canvas.palette.error;
            return {
                outer: isAccessible(palette.main) ? palette.main : palette.darkest,
                inner: palette.main,
            };
        }
        else {
            return {
                outer: inputColors.error.border,
                inner: inputColors.error.border,
            };
        }
    }
    else if (error === ErrorType.Alert) {
        if (theme) {
            var palette = theme.canvas.palette.alert;
            return {
                outer: isAccessible(palette.main) ? palette.main : palette.darkest,
                inner: palette.main,
            };
        }
        else {
            return {
                outer: colors.cantaloupe600,
                inner: inputColors.alert.border,
            };
        }
    }
    else {
        return {};
    }
}
export function errorRing(error, theme) {
    if (error !== ErrorType.Error && error !== ErrorType.Alert) {
        return {};
    }
    var errorColors = getErrorColors(error, theme);
    var errorBoxShadow = "inset 0 0 0 " + (errorColors.outer === errorColors.inner ? 1 : 2) + "px " + errorColors.inner;
    return {
        borderColor: errorColors.outer,
        transition: '100ms box-shadow',
        boxShadow: errorBoxShadow,
        '&:hover, &:disabled': {
            borderColor: errorColors.outer,
        },
        '&:focus:not([disabled])': {
            borderColor: errorColors.outer,
            boxShadow: errorBoxShadow + ",\n        0 0 0 2px " + colors.frenchVanilla100 + ",\n        0 0 0 4px " + (theme ? theme.canvas.palette.common.focusOutline : inputColors.focusBorder),
        },
    };
}
