// @workday/canvas-kit-labs-react-core/dist/commonjs/lib/type.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
var canvas_colors_web_1 = require("@workday/canvas-colors-web");
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var hierarchy = {
    brand1: {
        fontSize: 56,
        lineHeight: 1.3,
        fontWeight: 700,
        color: canvas_colors_web_1.typeColors.heading,
    },
    brand2: {
        fontSize: 48,
        lineHeight: 1.3,
        fontWeight: 700,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h1: {
        fontSize: 40,
        lineHeight: 1.3,
        fontWeight: 500,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h2: {
        fontSize: 32,
        lineHeight: 1.3,
        fontWeight: 700,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h3: {
        fontSize: 24,
        lineHeight: 1.3,
        fontWeight: 700,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h4: {
        fontSize: 20,
        lineHeight: 1.3,
        fontWeight: 700,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h5: {
        fontSize: 20,
        lineHeight: 1.3,
        fontWeight: 400,
        color: canvas_colors_web_1.typeColors.heading,
    },
    body: {
        fontSize: 16,
        lineHeight: 1.5,
        fontWeight: 400,
        color: canvas_colors_web_1.typeColors.body,
    },
    body2: {
        fontSize: 14,
        lineHeight: 1.5,
        fontWeight: 400,
        color: canvas_colors_web_1.typeColors.body,
    },
    small: {
        fontSize: 13,
        lineHeight: 1.5,
        fontWeight: 400,
        color: canvas_colors_web_1.typeColors.body,
    },
};
Object.keys(hierarchy).forEach(function (key) {
    hierarchy[key] = __assign(__assign({}, hierarchy[key]), { fontFamily: canvas_kit_react_core_1.fontFamily });
});
var updatedVariants = {
    button: {
        fontWeight: 600,
    },
    caps: {
        fontWeight: 500,
        letterSpacing: 0.5,
        textTransform: 'uppercase',
    },
};
exports.default = __assign(__assign({}, hierarchy), { variant: __assign(__assign({}, canvas_kit_react_core_1.type.variant), updatedVariants) });
